internal/config/config.go

```diff
@@ -57,35 +67,93 @@ type Config struct {
 	SidecarRequestsMemory string            `default:"40Mi" desc:"Lower bound of the NSM sidecar requests memory limits (in k8s resource management units)" split_words:"true"`
 	SidecarRequestsCPU    string            `default:"100m" desc:"Lower bound of the NSM sidecar requests CPU limits (in k8s resource management units)" split_words:"true"`
 	envs                  []corev1.EnvVar
+	secretsClient         coreV1Types.SecretInterface
 	caBundle              []byte
 	cert                  tls.Certificate
+	mode                  Mode
```

Comment:
Could be simplified, please have a look at comment above.
```suggestion
```

Reply:
Done